pkg/apis/internal/gs/state.go

```diff
@@ -0,0 +1,28 @@
+package gs
+
+import (
+	"go.anx.io/go-anxcloud/pkg/api/types"
+)
+
+// State represents the state object available on GS resources
+type State struct {
+	// programatically usable enum value
+	ID string `json:"id"`
+
+	// human readable status text
+	Text string `json:"text"`
+
+	Type int `json:"type"`
+}
+
+type objectWithStateRetriever interface {
+	types.Object
+	StateRetriever
+}
+
+// StateRetriever is an interface Objects can implement to provide unified state information
+type StateRetriever interface {
+	StateSuccess() bool
+	StateProgressing() bool
+	StateFailure() bool
```

Comment:
In theory, matching the Engine, we should rename those to `StateOK`, `StatePending` and `StateError`

Let's change this and not care about breaking users of go-anxcloud, should only be CCM anyway and is an easy fix